src/wallet/rpcwallet.cpp

```diff
@@ -364,15 +364,10 @@ UniValue getaddressesbyaccount(const UniValue &params, bool fHelp)
 
 static void SendMoney(const CTxDestination &address, CAmount nValue, bool fSubtractFeeFromAmount, CWalletTx &wtxNew)
 {
-    CAmount curBalance = pwalletMain->GetBalance();
-
     // Check amount
     if (nValue <= 0)
         throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid amount");
 
-    if (nValue > curBalance)
-        throw JSONRPCError(RPC_WALLET_INSUFFICIENT_FUNDS, "Insufficient funds");
-
```

Comment:
why removing this check? I mean if you don't have the amount you want to spend available what's the point to move further?

Reply:
This is actually an extremely expensive check that forces the wallet to iterate through every unspent coin.  In the non-errored case, its much less expensive to find a solution than it is to check the balance.  Only when the solution requires every unspent (or more -- ie "insufficient funds") does the coin selection perform as poorly as the balance check.  This error is discovered and reported at that point.

Reply:
I see. Thanks for the explanation